### ISSUE TITLE
chore(clippy): audit cast_* family workspace allows (cast_possible_wrap / cast_possible_truncation / cast_precision_loss)

### ISSUE THREAD

**Issue by maratik123** [maratik123/quartzite#443]
## Current state

Three workspace-level allows in root \`Cargo.toml [workspace.lints.clippy]\`, all justified by *"geometry & rendering deliberately cross between integer/float types within design-bounded ranges"*:

\`\`\`toml
# Geometry crate intentionally crosses i32↔u32 with design-bounded ranges; per-cast `try_from` would obscure intent
cast_possible_wrap = "allow"
# Geometry & rendering deliberately truncate within known bounds; per-call `try_from` adds noise without catching real bugs
cast_possible_truncation = "allow"
# i32 → f32 widening in geometry is intentional (sub-pixel coords); precision loss is the API contract
cast_precision_loss = "allow"
\`\`\`

Introduced in #423 (\`tighten-clippy-pedantic-nursery\`).

## Survey recipe

\`\`\`bash
# Temporarily comment out the three allows in root Cargo.toml, then:
cargo clippy --workspace --message-format=short 2>&1 \
  | rg '(cast_possible_wrap|cast_possible_truncation|cast_precision_loss)' \
  | awk -F: '{ print \$1 }' \
  | sort -u
\`\`\`

Record per-lint hit counts and per-file distribution.

## Audit questions

1. Is the *"geometry & rendering"* scope claim **actually accurate** — are all current hits in \`quartzite-geometry\` / \`quartzite-paint\` / \`quartzite-paint-api\` / \`quartzite-renderer\` / \`quartzite-style\`? If hits exist outside that scope, the allow may be silencing real bugs.
2. Can the allow be **narrowed** to those crates' \`[lints.clippy]\` blocks via per-crate config (\`<crate>/Cargo.toml\`), restoring the lint workspace-wide for other crates?
3. Can affected sites be **rewritten** using \`u32::try_from(…).unwrap_or(0)\` / \`.expect(\"…\")\` / explicit saturating-casts / \`as_-style\` helpers (\`as_f32()\` etc.) without losing readability?
4. Are there \`as\`-casts in the surveyed sites that **could be unsafe** (lossy under input the API contract doesn't actually bound)? Mass-allow forecloses that question per site.

## Decision (one of)

- **Keep** — the *"geometry & rendering"* scope claim is verified; the allow is genuinely load-bearing across enough sites that per-site \`#[allow]\` would be net noise. Close \`wontfix\` after updating the Cargo.toml comment with the surveyed hit count.
- **Narrow** — move the three allows from root \`Cargo.toml\` to the actual geometry/rendering crates' per-crate \`[lints.clippy]\` blocks. Verify other crates compile clean under the restored workspace lint.
- **Fix** — refactor protected sites to use explicit conversion helpers (e.g., \`i32_to_f32_pixel(x: i32) -> f32\` helpers, or \`saturating_cast!\` macros) so the cast intent is named at every site; remove the three workspace allows.

## Acceptance criteria

- [ ] Survey recipe executed; per-lint hit count and per-file distribution recorded in the PR body.
- [ ] Decision made between Keep / Narrow / Fix.
- [ ] If Keep: \`Cargo.toml\` comments updated to reflect surveyed scope + hit count.
- [ ] If Narrow: per-crate \`[lints.clippy]\` blocks in the affected crates; workspace allows removed; \`cargo clippy --workspace -- -D warnings\` exit 0.
- [ ] If Fix: protected sites refactored; workspace allows removed; \`cargo clippy --workspace -- -D warnings\` exit 0.

## Related

- #423 — \`tighten-clippy-pedantic-nursery\` (origin of the allows).
- AGENTS.md *Linter posture* — strict clippy; workspace lints in root \`Cargo.toml\`.

**Comment by maratik123** [comment]
Merged into #477 — all 11 clippy-allow audits consolidated into a single /task.
